certora/ArrangerConduit.spec

```diff
@@ -0,0 +1,805 @@
+
+    assert lastReverted <=> revert1 || revert2 || revert3 ||
+                            revert4, "Revert rules failed";
+} 
+
+// Verify correct storage changes for non reverting requestFunds
+rule cancelFundRequest(uint256 fundRequestId) {
+    env e;
+
+    uint256 anyIndex;
+
+    IArrangerConduit.FundRequest requestBefore = getFundRequest(anyIndex);
+    bytes32 infoHashBefore = aux.hashString(requestBefore.info);
+
+    mathint requestedFundsBefore = requestedFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalRequestedFundsBefore = totalRequestedFunds(requestBefore.asset);
+    mathint numRequestsBefore = getFundRequestsLength();
+
+    bytes32 otherIlk;
+    address otherAsset;
+    require otherIlk != requestBefore.ilk || otherAsset != requestBefore.asset;
+    mathint requestedFundsOtherBefore = requestedFunds(otherAsset, otherIlk);
+
+    address otherAsset2;
+    require otherAsset2 != requestBefore.asset;
+    mathint totalRequestedFundsOtherBefore = totalRequestedFunds(otherAsset2); 
+
+    cancelFundRequest(e, fundRequestId);
+
+    mathint requestedFundsAfter = requestedFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalRequestedFundsAfter= totalRequestedFunds(requestBefore.asset);
+    mathint numRequestsAfter = getFundRequestsLength();
+
+    mathint requestedFundsOtherAfter = requestedFunds(otherAsset, otherIlk);
+    mathint totalRequestedFundsOtherAfter = totalRequestedFunds(otherAsset2); 
+
+    IArrangerConduit.FundRequest requestAfter = getFundRequest(anyIndex);
+    bytes32 infoHashAfter = aux.hashString(requestAfter.info);
+
+    assert numRequestsAfter == numRequestsBefore, "num requests changed";
+    assert anyIndex == fundRequestId => requestedFundsAfter == requestedFundsBefore - requestBefore.amountRequested, "cancelFundRequest did not decrease by amount";
+    assert anyIndex == fundRequestId => totalRequestedFundsAfter == totalRequestedFundsBefore - requestBefore.amountRequested, "totalRequestedFunds did not decrease by amount";
+    assert anyIndex == fundRequestId => requestAfter.status == IArrangerConduit.StatusEnum.CANCELLED, "cancelFundRequest did not change status to CANCELLED";
+    assert anyIndex != fundRequestId => requestAfter.status == requestBefore.status, "cancelFundRequest on another index changed status";
+    assert requestAfter.asset == requestBefore.asset
+        && requestAfter.ilk == requestBefore.ilk
+        && requestAfter.amountRequested == requestBefore.amountRequested
+        && requestAfter.amountFilled == requestBefore.amountFilled
+        && infoHashAfter == infoHashBefore,
+        "other request params not as before";
+
+    assert anyIndex == fundRequestId => requestedFundsOtherAfter == requestedFundsOtherBefore, "other requested funds changed unexpectedly";
+    assert anyIndex == fundRequestId => totalRequestedFundsOtherAfter == totalRequestedFundsOtherBefore, "other total requested funds changed unexpectedly";
```

Comment:
It's a bit weird how this thing ended up. Maybe is easier to think if we would have a `requestBefore/After` using the `fundRequestId` and then a `requestOtherBefore/After` using `anyIndex` which should be `!= fundRequestId`.
That way it would be more inline how we treat other storages.

Reply:
Implemented here - 
https://github.com/makerdao/dss-conduits/pull/18

These rules are quite heavy on the eye, we can consider separating to several ones.